CollapseLauncher/Classes/FileDialogCOM/FileDialogNative.cs

```diff
@@ -1,199 +1,280 @@
-﻿using CollapseLauncher.FileDialogCOM;
+using Hi3Helper;
 using System;
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
 using System.Threading.Tasks;
+using Windows.Win32;
 
 namespace CollapseLauncher.FileDialogCOM
 {
-    // Reference:
-    // https://www.dotnetframework.org/default.aspx/4@0/4@0/DEVDIV_TFS/Dev10/Releases/RTMRel/ndp/fx/src/WinForms/Managed/System/WinForms/FileDialog_Vista_Interop@cs/1305376/FileDialog_Vista_Interop@cs
+    /*
+     * Reference:
+     * https://www.dotnetframework.org/default.aspx/4@0/4@0/DEVDIV_TFS/Dev10/Releases/RTMRel/ndp/fx/src/WinForms/Managed/System/WinForms/FileDialog_Vista_Interop@cs/1305376/FileDialog_Vista_Interop@cs
+     * 
+     * UPDATE: 2023-11-01
+     * This code has been modified to support ILTrimming and Native AOT
+     * by using Source-generated COM Wrappers on .NET 8.
+     * 
+     * Please refer to this link for more information:
+     * https://learn.microsoft.com/en-us/dotnet/standard/native-interop/comwrappers-source-generation
+     */
     public static class FileDialogNative
     {
         private static IntPtr parentHandler = IntPtr.Zero;
         public static void InitHandlerPointer(IntPtr handle) => parentHandler = handle;
 
-        public static async Task<List<string>> GetMultiFilePicker(Dictionary<string, string> FileTypeFilter = null) => await Task.Run(() =>
+        public static async Task<string[]> GetMultiFilePicker(Dictionary<string, string> FileTypeFilter = null) => await Task.Run(() =>
         {
-            IFileOpenDialog dialog = null;
+            PInvoke.CoCreateInstance(
+                new Guid(CLSIDGuid.FileOpenDialog),
+                null,
+                Windows.Win32.System.Com.CLSCTX.CLSCTX_INPROC_SERVER,
+                out IFileOpenDialog dialog).ThrowOnFailure();
+
             IShellItemArray resShell;
 
             try
             {
-                dialog = new NativeFileOpenDialog();
                 dialog.SetOptions(FOS.FOS_NOREADONLYRETURN | FOS.FOS_DONTADDTORECENT | FOS.FOS_ALLOWMULTISELECT);
-                SetFileTypeFiler(ref dialog, FileTypeFilter);
+                SetFileTypeFilter(dialog, FileTypeFilter);
 
                 if (dialog.Show(parentHandler) < 0) return null;
 
                 dialog.GetResults(out resShell);
-                return GetIShellItemArray(dialog, resShell);
+                return GetIShellItemArray(resShell);
             }
             catch (COMException)
             {
                 return null;
             }
+#if !NET8_0_OR_GREATER
```

Comment:
Yes, we will be dropping .NET 7 support entirely and move to .NET 8 since COM Interop is only available on .NET 8 so this pre-define can be removed.
Will do the favor once I'm back home.

Reply:
As per latest commit, the defines have been deleted